dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/PoolingLayer.java

```diff
@@ -102,12 +163,54 @@ public Matrix feedForward(final Matrix input) {
     }
   }
 
+  /**
+   * Backpropagating function for max pooling.
+   * @param input the input values for this layer.
+   * @param nextError the errors of the next layer - the one closer to the output layer.
+   * @return errors for this layer with the specified input value.
+   */
   private Matrix backPropagateMaxPooling(final Matrix input, final Matrix nextError) {
-    throw new RuntimeException("Not implemented");
+    final Matrix error = matrixFactory.zeros(input.getRows(), input.getColumns());
+    for (int n = 0; n < input.getColumns(); ++n) {
+      for (int oh = 0; oh < outputShape[0]; ++oh) {
+        for (int ow = 0; ow < outputShape[1]; ++ow) {
+          //Add error to saved index.
+          final int index = oh * outputShape[1] + ow;
+          final float tempError = nextError.get(index, n) + error.get(index, n);
```

Comment:
I'm wondering if `error.get(index, n)` is correct. Is it `error.get((int) indexMatrix.get(index, n), n)`?